integration-tests/deploy.py

```diff
@@ -0,0 +1,121 @@
+            response = request_metrics(f"800{item}")
+            if response.status_code != 200:
+                print(f'Failed to get response from bob instance {item}, retrying...')
+                raise UserWarning(f'Timeout on getting response from node {item}.')
+            else:
+                metrics = json.loads(response.content.decode('ascii'))
+                if metrics['metrics']['backend.backend_state']['value'] != 1:
+                    print(f'Backend is not up on node {item}, waiting...')
+                    raise UserWarning(f'Backend is down on node {item}.')
+                else:
+                    print(f'Node {item} is ready!')
+        print('All nodes are ready!')
+    except KeyError:
+        sys.exit('Nodes amount is not set.')
+    except ValueError:
+        sys.exit('Amount of nodes has unexpected value.')
+    
+
+#collect arguments
+parser = argparse.ArgumentParser(description='Deploys docker compose nodes.')
+
+parser.add_argument('--path', dest='path', type=str, required=True, help='Takes in path to generated configs.')
+parser.add_argument('-r', dest='replicas', type=int, required=True, help='Sets amount of replicas to create in cluster.')
+exclusive = parser.add_mutually_exclusive_group(required=True)
+exclusive.add_argument('-d', dest='vdisks_count', nargs='?', type=int, help='min - equal to number of pairs node-disk.')
+exclusive.add_argument('-p', dest='vdisks_per_disk', nargs='?', type=int, help='number of vdisks per physical disk.')
+
+args = parser.parse_args()
+final_args = {'-r':args.replicas, '-d':args.vdisks_count, '-p':args.vdisks_per_disk}
+
+#transform args to a string format
+args_str = str()
+for (key) in final_args:
+    if final_args.get(key) != None:
+        args_str += f'{key} {final_args.get(key)} '
+
+#initilize docker client
+client = docker.from_env()
+
+#check for network existing and creating it
+try:
+    bobnet = client.networks.get('bob_net')
+except d_err.NotFound:
+    ipam_config = d_types.IPAMConfig(pool_configs=[d_types.IPAMPool(subnet='172.21.0.0/24', gateway='172.21.0.1')])
+    bobnet = client.networks.create('bob_net', driver='bridge', ipam=ipam_config, attachable=True)
+
+start_path = os.getcwdb()
+good_path = os.path.abspath(args.path)
+
+#check for configs to exist
+try:
+    if not 'cluster.yaml.bobnet' in os.listdir(good_path):
+        sys.exit('Cluster config not found in the specifed directory.')
+except FileNotFoundError as e:
+    sys.exit(e)
+
+#change execution permissions for binaries
+try:
+    os.chmod(path=f'./ccg', mode=0o771)
+    os.chmod(path=f'./bobp', mode=0o771)
+except OSError as e:
+    sys.exit(e)
+
+#run cluster generation
+try:
+    pr = subprocess.check_output(shlex.split(f'./ccg new -i {args.path}/cluster.yaml.bobnet -o {args.path}/cluster.yaml.bobnet {args_str.rstrip()}'))
+    if str(pr).find('ERROR') != -1:
+        sys.exit(str(pr))
+except subprocess.CalledProcessError:
+    sys.exit(pr.stderr)
```

Comment:
I don't know python well, but looks like `pr` is out of scope here, so there will not be access to `stderr`

Reply:
I checked it, it is in scope, stderr is handled correctly